src/provider/application.js

```diff
@@ -121,19 +133,24 @@ class Application extends EventEmitter {
       // 2: Begin launch and authorization sequence
       this.JSONRPC.notification('launch');
 
-      // 2a. We have a specific origin to trust (excluding wildcard *),
-      // wait for response to authorize.
-      if (this.acls.some((x) => x !== '*')) {
-        this.JSONRPC.request('authorizeConsumer', [])
-          .then(this.authorizeConsumer)
-          .catch(this.emitError);
-      }
-
-      // 2b. We don't know who to trust, challenge parent for secret
-      if (this.secret) {
-        this.JSONRPC.request('challengeConsumer', [])
-          .then(this.verifyChallenge)
-          .catch(this.emitError);
+      if (this.autoAuthorizeConsumer) {
```

Comment:
this if condition can be merged into above if condition for `(window.self !== window.top)` in line 129.

Reply:
I think we still want these two steps to happen:

```
// 1: Setup listeners for all incoming communication
window.addEventListener('message', this.handleConsumerMessage);
 
// 2: Begin launch and authorization sequence
this.JSONRPC.notification('launch');
```

Reply:
ah, I see. make sense.